### ISSUE TITLE
Better Documentation for hs.fs.dir()

### ISSUE THREAD

**Issue by latenitefilms** [Hammerspoon/hammerspoon#1927]
Based on the API documentation it's not clear that `hs.fs.dir()` will return a `nil` value for the iterator function if something doesn't work. For example:

```
> iterFn, dirObj = hs.fs.dir("/Users/Guest/Documents")
print(string.format("iterFn: %s", iterFn))
print(string.format("dirObj: %s", dirObj))
2018-08-28 12:11:11: iterFn: nil
2018-08-28 12:11:11: dirObj: cannot open /Users/Guest/Documents: Permission denied
```

This becomes problematic when you're using a `for` loop, for example:

```
for file in hs.fs.dir("/Users/Guest/Documents") do 
	print(file)
end
```

Maybe either the documentation or the function itself needs to be re-thought @cmsj or @asmagill ?

**Comment by cmsj** [comment]
I suspect probably the docs should include an example

**Comment by latenitefilms** [comment]
Fixed in #1928
